frontend/packages/kubevirt-plugin/integration-tests/tests/vm.template.wizard.scenario.ts

```diff
@@ -100,11 +100,12 @@ describe('Kubevirt create VM Template using wizard', () => {
         );
         const vm = new VirtualMachine(vmConfig(configName.toLowerCase(), vmTemplate));
 
-        addLeakableResource(leakedResources, vmTemplate.asResource());
-        addLeakableResource(leakedResources, vm.asResource());
-
-        await vmTemplate.create(vmTemplateConfig(configName.toLowerCase(), provisionConfig));
-        await vm.create(vmConfig(configName.toLowerCase(), vmTemplate));
+        await withResource(leakedResources, vmTemplate.asResource(), async () => {
```

Comment:
Just wondering, does it make sense to have `withResources` (plural) that takes variable number of resources (var-arg) or an array of resources?

```ts
await withResources(leakedResources, [vmTemplate.asResource(), vm.asResource()], async () => {
  await vmTemplate.create(vmTemplateConfig(configName.toLowerCase(), provisionConfig));
  await vm.create(vmConfig(configName.toLowerCase(), vmTemplate));
});
```

Also, if we're set on using `withResource`(s) function, shouldn't we also use it on other places, effectively replacing `addLeakableResource` calls?

Reply:
1)
It looks nice, we would need to improve the `withResource` logic.
In this example, if something happened (exception, timeout) during the `vmTemplate` creation, `withResource` would attempt to remove `vm` as well, but it would not be created.
We could add a property to the objects that would indicate whether or not this object has already been instantiated in the cluster, `withResource` would then check this property before removing it.
2) Regarding `addLeakableResource`, these methods are used in combination with `beforeAll/Each, afterAll/Each`, I'm not sure if we can use `withResource` in these situations

Reply:
@rhrazdil Thanks for the clarification. I'm OK with current code (sounds like my suggestion would yield more code complexity).